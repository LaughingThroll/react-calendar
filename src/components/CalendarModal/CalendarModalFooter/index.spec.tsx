import { shallow } from 'enzyme'
import { CalendarModalFooter, CalendarModalFooterProps } from './'

const setUp = (props: CalendarModalFooterProps) => shallow(<CalendarModalFooter {...props} />)

describe('CalendarModalFooter render', () => {
  it('default render', () => {
    const props = { onClose: jest.fn(), onSubmit: jest.fn(), isDisabled: false }
    const tree = setUp(props)

    expect(tree).toMatchSnapshot()
  })
})

describe('CalendarModalFooter test props', () => {
  it('must be call fn onClose', () => {
    const props = { onClose: jest.fn(), onSubmit: jest.fn(), isDisabled: false }
    const tree = shallow(<CalendarModalFooter {...props} />)

    tree.children().first().simulate('click')

    expect(props.onClose).toHaveBeenCalled()
    expect(props.onSubmit).toHaveBeenCalledTimes(0)
  })

  it('must be call fn onSubmit', () => {
    const props = { onClose: jest.fn(), onSubmit: jest.fn(), isDisabled: false }
    const tree = shallow(<CalendarModalFooter {...props} />)

    tree.children().last().simulate('click')

    expect(props.onSubmit).toHaveBeenCalled()
    expect(props.onClose).toHaveBeenCalledTimes(0)
  })

  it('must be ignore fn onSubmit', () => {
    const props = { onClose: jest.fn(), onSubmit: jest.fn(), isDisabled: true }
    const tree = shallow(<CalendarModalFooter {...props} />)
    const $submitButton = tree.children().last()

    $submitButton.simulate('click')

    expect(props.onSubmit).toHaveBeenCalledTimes(0)
    expect(props.onClose).toHaveBeenCalledTimes(0)
  })
})
