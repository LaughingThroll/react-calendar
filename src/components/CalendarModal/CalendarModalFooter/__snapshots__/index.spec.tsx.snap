// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarModalFooter render default render 1`] = `
<Memo(ModalFooter)
  className={Array []}
>
  <Memo(Button)
    onClick={[MockFunction]}
    secondary={true}
  >
    Cancel
  </Memo(Button)>
  <Memo(Button)
    disabled={false}
    onClick={[MockFunction]}
    type="submit"
  >
    Send
  </Memo(Button)>
</Memo(ModalFooter)>
`;
